File: src/com/basmilius/ps/bastools/action/ideemedia/angular/VersionAction.java
package com.basmilius.ps.bastools.action.ideemedia.angular;

import com.basmilius.ps.bastools.util.CommandUtils;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;

import java.util.ArrayList;
import java.util.List;

public class VersionAction extends AnAction
{

	public VersionAction ()
	{
		super();
	}

	@Override
	public void actionPerformed (final AnActionEvent aae)
	{
		final Project project = aae.getProject();

		if (project == null)
		{
			return;
		}

		CommandUtils.run(project, "powershell ng version", lines ->
		{
			final List<String> result = new ArrayList<>();

			for (final String line : lines)
			{
				if (line.startsWith("@angular/") || line.startsWith("node:") || line.startsWith("os:"))
				{
					result.add(line);
				}
			}

			ApplicationManager.getApplication().invokeLater(() -> Messages.showMessageDialog(project, String.join(System.getProperty("line.separator"), result), "Angular Version", Messages.getInformationIcon()));
		});
	}

}

File: src/com/basmilius/ps/bastools/action/angular/VersionAction.kt
package com.basmilius.ps.bastools.action.angular

import com.basmilius.ps.bastools.util.CommandUtils
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.ui.Messages

/**
 * Class VersionAction
 *
 * @author Bas Milius
 * @package com.basmilius.ps.bastools.action.angular
 */
class VersionAction : AnAction("Show Angular Version")
{

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	override fun actionPerformed(aae: AnActionEvent)
	{
		val project = aae.project ?: return

		CommandUtils.run(project, "powershell ng version", { lines: Array<String> ->
			val result = lines.filter { it.startsWith("@angular/") || it.startsWith("node:") || it.startsWith("os:") }

			ApplicationManager.getApplication().invokeLater { Messages.showMessageDialog(project, result.joinToString(System.getProperty("line.separator")), "Angular Version", Messages.getInformationIcon()) }
		} as CommandUtils.ICommandResultHandler)

	}

}
